src/ItemsStorage.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Rbac\Cycle;
+
+use Cycle\Database\DatabaseInterface;
+use Cycle\Database\DatabaseProviderInterface;
+use Cycle\Database\Injection\Expression;
+use Cycle\Database\Table;
+use Cycle\Database\TableInterface;
+use Yiisoft\Rbac\Item;
+use Yiisoft\Rbac\ItemsStorageInterface;
+use Yiisoft\Rbac\Permission;
+use Yiisoft\Rbac\Role;
+
+final class ItemsStorage implements ItemsStorageInterface
+{
+    private DatabaseInterface $database;
+    private Table|TableInterface $table;
+    private Table|TableInterface $childrenTable;
```

Comment:
Here can be any implementation of `TableInterface` (non-`Table` class)? 
Many psalm errors because of this. For example, `TableInterface` don't contain `select()` method.

Reply:
> Here can be any implementation of TableInterface (non-Table class)?

I figured out the cycle and there is no implementation other than the `Table` class.